index.js

```diff
@@ -12,17 +12,31 @@ const GitHubButton = React.memo(({ children, ...props }) => {
 
     import(/* webpackMode: "eager" */ "github-buttons")
       .then(({ render }) => {
-        if (buttonRef.current && containerRef.current) {
+        if (
+          buttonRef.current &&
+          containerRef.current &&
+          containerRef.current.lastChild === tempSpan
+        ) {
           render(tempSpan.appendChild(buttonRef.current), (el) => {
```

Comment:
Initial state:
``` html
<span> <!-- containerRef-->
  <a> <!-- buttonRef -->
  </a>
</span>
```

After adding a `tempSpan`:

``` html
<span> <!-- containerRef-->
  <a></a> <!-- buttonRef -->
  <span></span> <!-- tempSpan -->
</span>
```

With `tempSpan.appendChild(buttonRef.current)`: 

``` html
<span> <!-- containerRef-->
  <span> <!-- tempSpan -->
    <a></a> <!-- buttonRef -->
  </span>
</span>
```

Therefore, when restoring the state on error or inside `useEffect`, it should be:

``` js
containerRef.current.replaceChild(buttonRef.current, containerRef.current.lastChild);
```

Reply:
The reason we have this `tempSpan` is because this library runs async rendering that dispatches outside of react lifecycles, therefore we want to make sure when we come back to react lifecycle, react should not see any DOM patch that we have done so that it can patch DOM incrementally. In order to do that we have to restore the DOM to the state that react generates. The primary goal of using an additional `tempSpan` is allow us to easily check if we have patched the DOM outside of react or not, and acting accordingly.